sources/Renderer/OpenGL/RenderState/GLStateManager.cpp

```diff
@@ -550,6 +550,18 @@ void GLStateManager::SetLineWidth(GLfloat width)
     }
 }
 
+void GLStateManager::SetPrimitiveRestartIndex(bool indexType16Bits)
+{
+    #ifdef LLGL_PRIMITIVE_RESTART
+    GLuint index = (indexType16Bits ? 0xFFFF : 0xFFFFFFFF);
+    if (commonState_.primitiveRestartIndex != index)
+    {
+        commonState_.primitiveRestartIndex = index;
+        glPrimitiveRestartIndex(index);
```

Comment:
This function looks good, but we'll have to add `glPrimitiveRestartIndex` to the declaration of GL extensions (see [GLCoreExtensionsDecl.inc](https://github.com/LukasBanana/LLGL/blob/master/sources/Renderer/OpenGL/GLCoreProfile/GLCoreExtensionsDecl.inl)). This is not necessary for MacOS, but for the other platforms. I'm sometimes having a hard time finding the right extension either on the [OpenGL registry page](https://www.khronos.org/registry/OpenGL/index_gl.php) or via a search engine. If you can't find it either, maybe you can add another section for `GL_VERSION_3_1` like the one for [`glGetStringi`](https://github.com/LukasBanana/LLGL/blob/master/sources/Renderer/OpenGL/GLCoreProfile/GLCoreExtensionsDecl.inl#L66).

If you need help with that part, I can take care of it after I merged the PR.

Reply:
From what I can tell primitive restart went straight from an NV extension (GL_NV_PRIMITIVE_RESTART) to 3.1 core. I took a shot at updating this file but don't have a good way of testing my changes, I only code on Mac and Raspberry Pi these days.